src/net/larse/lcms/algorithms/VCT.java

```diff
@@ -1029,35 +841,35 @@ private int getMaxForLength(double minForUd, double maxForUd) {
     int waterCount = 0;
     int maxLength = 0;
 
-
-    //TODO: (yang) verify this single for-loop to replace the double while loop,
-//    int currentWaterCount = 0;
-//    int currentLength = 0;
-//    for (int i = 0; i < this.numYears; i++) {
-//      double tmp = this.ud[COMP][i];
-//      if (tmp <= maxForUd && tmp >= minForUd) {
-//        currentLength++;
-//        if (this.mask[i] == WATER) {
-//          currentWaterCount++;
-//        }
-//      }
-//      else {
-//        if (maxLength < currentLength) {
-//          maxLength = currentLength;
-//          waterCount = currentWaterCount;
-//          iEnd = i - 1;
-//        }
-//        currentWaterCount = 0;
-//        currentLength = 0;
-//        iStart = i;
-//      }
-//    }
-//    //if the last segment is the longest
-//    if (maxLength < currentLength) {
-//      maxLength = currentLength;
-//      waterCount = currentWaterCount;
-//      iEnd = this.numYears - 1;
-//    }
+    // TODO: (yang) verify this single for-loop to replace the double
+    // while loop
+    // int currentWaterCount = 0;
+    // int currentLength = 0;
+    // for (int i = 0; i < this.numYears; i++) {
+    //   double tmp = this.ud[COMP][i];
+    //   if (tmp <= maxForUd && tmp >= minForUd) {
+    //     currentLength++;
+    //     if (this.mask[i] == WATER) {
+    //       currentWaterCount++;
+    //     }
+    //   }
+    //   else {
+    //     if (maxLength < currentLength) {
+    //       maxLength = currentLength;
+    //       waterCount = currentWaterCount;
+    //       iEnd = i - 1;
+    //     }
+    //     currentWaterCount = 0;
+    //     currentLength = 0;
+    //     iStart = i;
+    //   }
+    // }
+    // // if the last segment is the longest
+    // if (maxLength < currentLength) {
+    //   maxLength = currentLength;
+    //   waterCount = currentWaterCount;
+    //   iEnd = this.numYears - 1;
+    // }
```

Comment:
@c11, did you want to use this code instead of following while loops.  I haven't tested this.